lib/widgets/add_expense.dart

```diff
@@ -0,0 +1,124 @@
+          padding: const EdgeInsets.all(16.0),
+          width: double.infinity,
+          child: SingleChildScrollView(
+            child: ListBody(children: [
+              TextFormField(
+                textAlign: TextAlign.center,
+                controller: _expenseValueController,
+                keyboardType: TextInputType.number,
+                inputFormatters: <TextInputFormatter>[
+                  FilteringTextInputFormatter.digitsOnly
+                ],
+                validator: (value) {
+                  const _validationError = 'Enter a valid expense value';
+                  if (value!.isEmpty) {
+                    return _validationError;
+                  }
+                  if (double.parse(value) <= 0) {
+                    return _validationError;
+                  }
+                  return null;
+                },
+                style: const TextStyle(fontSize: 70, color: Colors.black),
+                autofocus: true,
+                autocorrect: false,
+                decoration: const InputDecoration(hintText: "0.00"),
+                textInputAction: TextInputAction.next,
+              ),
+              const SizedBox(
+                height: 20.0,
+              ),
+              TextFormField(
+                controller: _expenseNameController,
+                autofocus: true,
+                autocorrect: false,
+                decoration: const InputDecoration(hintText: "Expense Name"),
+                textInputAction: TextInputAction.done,
+                validator: (name) {
+                  if (name!.isNotEmpty) {
+                    return null;
+                  } else {
+                    return 'Enter a valid expense name';
+                  }
+                },
+              ),
+              const SizedBox(
+                height: 20.0,
+              ),
+              FutureBuilder<List<ExpenseCategory?>?>(
+                  future: apiService.getExpenseCategories(),
+                  builder: (context, snapshot) {
+                    return DropdownButtonFormField<ExpenseCategory>(
+                      hint: const Text("Select"),
+                      onChanged: (newValue) {
+                        _selectedExpenseCategory = newValue!;
+                      },
+                      // value: selectedCategory,
+                      items: snapshot.data
+                          ?.map((ec) => DropdownMenuItem<ExpenseCategory>(
+                                value: ec,
+                                child: Text(ec!.categoryname),
+                              ))
+                          .toList(),
+                    );
+                  }),
+              const SizedBox(
+                height: 20.0,
+              ),
+            ]),
+          ),
+        ),
+      ),
+      actions: [
+        TextButton(
+            child: const Text('OK'),
+            onPressed: () async {
+              if (formGlobalKey.currentState!.validate()) {
```

Comment:
question: Does `currentState` need to be null checked here? Dk if it might really be null at some point.